Updated merge of old repos with tests added

tacketar opened the pull request:
This is massive and unwieldy pull request.  Getting this merged in should get us back to making more manageable pull requests.  This has a litany of changes:
- All changes/bugfixes added pre-refactor have been incorporated.
- More name-mangling done to fit the naming scheme as I came across them.  There is more to do
- All old tests have been added to the tests directory and all bugs discovered fixed. This is **NOT** integrated into the testing framework.  That will be done at a later date
-  A lot of routines were added to the public API that were used by the test programs.  I don't think any new structures were exposed but several constructor/destructor/accessor functions were added to make objects usable outside of the core library.
- Several structs were moved out of the API.  These just had a few references and could be easily converted to ptrs.
- Resolves issue #126 and #127

tacketar committed:
Only include UP and NO_SPACE drives in the usable tally

tacketar committed:
Properly pass the error code back to the calling program

tacketar committed:
Add support for getting the file list from STDIN

tacketar committed:
Snag the segment ID before decrementing the flushing_count

The final log statement uses the segment ID which could be invalid
because once the flushing_count is decremented the segment_destroy
call could complete and destroy the actual segment.

tacketar committed:
Fix lio_inspect from segfaulting on SIGQUIT instead of cleanly winding down

tacketar committed:
Fix bug when renaming a file

tacketar committed:
Remove all the cruft lioc_* routines and rename lio_core.c->lio_core_misc.c

tacketar committed:
Fix missed variable initialization

tacketar committed:
Added support for LIO_EXCL_MODE when opening files

tacketar committed:
Properly lock and roll-back if needed rename operations.

tacketar committed:
Check and make sure we have a non-null pointer

tacketar committed:
Simplify the move object routine

tacketar committed:
Fix the move object function and propagate the error code back to FUSE

tacketar committed:
Fix issue on overwriting a file and initialize a status

tacketar committed:
Fix bug stopping depot updates from occuring.

tacketar committed:
Fix signed/unsigned int warnings

tacketar committed:
Start the standardization of collecting paths from the command line

tacketar committed:
Add using stock LevelDB to build

tacketar committed:
Add support for LevelDB to lio_warm and add warmer_query for mining

tacketar committed:
Cast the cap to a string for printing
This is not ideal there area couple of options 1) add a string helper function
2) just make the cap a printable string.  I don't think there is much an issue
doing this.

tacketar committed:
Switch to using the standard path iterator

tacketar committed:
Forgot to destroy the iterator

tacketar committed:
Need to strdup the rid_key otherwise you get a double free on cleanup

tacketar committed:
Fix indentation

tacketar committed:
Fix alphabetical order

tacketar committed:
Fix typos

tacketar committed:
Fix alphabetical order

tacketar committed:
Break up initialized and uninitialized variable declarations

tacketar committed:
Put objects in what I think is the proper format

tacketar committed:
Move the stdin/array iterator routines from lio->toolbox

tacketar committed:
Swtich the execs to using the toolbox stdin/array iterator

tacketar committed:
Remove the old stdin/array routines

tacketar committed:
Put the updated license

tacketar committed:
Adding old varint test. Not integrated with test suite yet

tacketar committed:
Add opaque helper functions for skip list iterators
I think all the *compare* functions should be deprecated at a later date.  Also need to
standardize on helper naming convention.  In any case there needs to be another round
of simplication.

tacketar committed:
Adding old skiplist test. Not integrated with test suite yet

tacketar committed:
Move unused attribute declaration to header file

tacketar committed:
Switch warmer DBs to use varints for portability

tacketar committed:
Cleanup junk

tacketar committed:
Properly munge routines that return a GOP to be named *_gop

tacketar committed:
Make older GCC compilers happy
Looks like older compilers think that lock_src and lock_dest can be
used uninitialized which is is impossible.

tacketar committed:
Add leveldb package dependencies for builds

tacketar committed:
Remove unused function

tacketar committed:
More name munging to fit new scheme

tacketar committed:
Removed old init calls

tacketar committed:
Switch to using new/del

tacketar committed:
Add functions for using iterators

tacketar committed:
Adding old ISL test program

tacketar committed:
Provide a more descriptive function name tbx_isl_count2()->tbx_isl_range_count()

tacketar committed:
Merge branch 'master' of https://github.com/accre/lstore

tacketar committed:
Merge branch 'master' into feature/tests
This incormporates Andrews changes

tacketar committed:
Fix mangles names: gop_gop_* -> gop_*

tacketar committed:
Initial version with mq_test before API changes

tacketar committed:
Expose the ability to set the max log size

tacketar committed:
Add helper for getting the stream id

tacketar committed:
Add mqs_test

tacketar committed:
Add mqs_test

tacketar committed:
Initial working versions of mq_test and mqs_test. Still need to mangle some names.
I had to move a bunch of functions to the API. Most of them should have definitely been
there from the beginning.  I did NOT change the names from mq_* gop_mq_*.  I'll
do it in a separate commit since that touches lots of code

tacketar committed:
Added proper name mangling

tacketar committed:
More name mangling

tacketar committed:
Move routine
ibp_sync_command is only used as part of the synchronous IBP API which is not in the base.
It does still exist in the test/ibp_sync.[ch] files for those that need them.

tacketar committed:
Update the license

tacketar committed:
Remove unneeded typedef

tacketar committed:
Move some functions to the public API

tacketar committed:
Expose the constructor/destructor routines so they can be used

tacketar committed:
Publish the structs
They are needed in order to write your own tbuf transfer function. You could possibly
write accessors but not sure if that's actually hiding things.

tacketar committed:
Expose IBP context accessor functions

tacketar committed:
Missed some files in all the sed-ing

tacketar committed:
Add constructor/destructor functions

tacketar committed:
Properly update symlinked missing attribute

tacketar committed:
Add tag to follow symlinks.  Defaults to skip them

tacketar committed:
Don't warm or follow symbolic links

tacketar committed:
Add additional accessor fns

tacketar committed:
Fix cut&paste type mistake

tacketar committed:
Move proxy capstatus routines to the public API
NOTE: This whole framework is deprecated and will be yanked soon.  As a result
I didn't properly mangle the names.

tacketar committed:
Add an ibp_ridlist_t constructor and export it

tacketar committed:
Remove extraneous dependency on the internals of ibp_op_t

tacketar committed:
Move all the ibp_op_t needed structs to be private

tacketar committed:
Remove no longer needed comment

tacketar committed:
Properly trap a failed ibp_rename operations

tacketar committed:
Trap partial return status and respond accordingly

tacketar committed:
Properly specify the operation type for an append op

tacketar committed:
Initial working version of ibp_test
Still need to handle memory leaks

tacketar committed:
Switch the sync calls to use the direct submission type connections

tacketar committed:
Simplify failure logic to get proper cleanup

tacketar committed:
Destroy the GOP after finished

tacketar committed:
Add extra capset constructor/destructor functions

tacketar committed:
Fix refactor bug resulting in a double allocation of the head node

tacketar committed:
Clean up iter when finished

tacketar committed:
All memory leaks fixed

tacketar committed:
Properly cleanup the jerase_paranoid object

tacketar committed:
Export the tbx_info_destroy destructor

tacketar committed:
Destroy the lio_ifd object

tacketar committed:
Properly destroy the old rid_table before overwriting

tacketar committed:
Track the replacement argv for freeing later

tacketar committed:
Destroy the attributes after printing

tacketar committed:
Destroy the LSE struct itself after printing

tacketar committed:
Forgot to free the filename

tacketar committed:
Do some simple error checking

tacketar committed:
Remove duplicate typedefs

tacketar committed:
Stop the arg processing loop from being optimized
Declaring the start_index as volatile disables the optimization. Without it
you get the following warning:

assuming signed overflow does not occur when assuming that (X + c) >= X is always true

on the 'while' condition

tacketar committed:
Fix memory leaks

tacketar committed:
Simplify the logic and correct appending / to dirs only

tacketar committed:
Update following symlink logic

tacketar committed:
Cleanup leaking the filename

tacketar committed:
Fix object refererence counting so segments are properly destroyed
The old method had the inc/dec and destruction as separate functions
unlike the new method, using tbx_{obj}ref}.h, where they are combined.
these are now combined as a single unit.

tacketar committed:
Remove blank lines

tacketar committed:
Drop duplicate typedefs and just put them in the public one.
It was creating issues compiling some executables.  Nothing extra is
being exposed.

tacketar committed:
Make structures private.
This is only used by lio_warm.  As a work around I added a hack function.
Once another DS is implemented we can extrapoalte and make a proper API
for warming.

tacketar committed:
Add OS testing routines

tacketar committed:
Added OS test program
The core tests were added below the API just exposing the high-level
tests which are then called form the actual test

tacketar committed:
Cleanup path
I have a very bad feeling all the lio_* executables need a similar patch

tacketar committed:
Fix lio_read and lio_myopen_fn
lio_read wasn't always returning the number of bytes actually read but instead the buffer size.
lio_myopen_fn had a failure edge case not propagating an error.  The fd was set to NULL properly
but not the status.

tacketar committed:
Move more of the POSIX-type functions to the API

tacketar committed:
Add the segment tester

PerilousApricot commented:
This PR doesn't merge against master, @tacketar 

```
hudson.plugins.git.GitException: Command "git merge 0458da62853f97893c51ab10e338be91e4e76239" returned status code 1:
stdout: Auto-merging src/toolbox/tbx/network.h
Auto-merging src/toolbox/tbx/list.h
CONFLICT (content): Merge conflict in src/toolbox/tbx/list.h
Auto-merging src/lio/lio_fuse_core.c
CONFLICT (content): Merge conflict in src/lio/lio_fuse_core.c
Auto-merging src/lio/lio_core_io.c
CONFLICT (content): Merge conflict in src/lio/lio_core_io.c
Auto-merging src/lio/lio_config.c
Auto-merging src/lio/lio/lio.h
CONFLICT (content): Merge conflict in src/lio/lio/lio.h
Auto-merging src/lio/lio.h
CONFLICT (content): Merge conflict in src/lio/lio.h
Auto-merging README.md
Auto-merging CMakeLists.txt
Automatic merge failed; fix conflicts and then commit the result.
```

tacketar committed:
Fix offset

tacketar committed:
Merge branch 'master' of https://github.com/accre/lstore

tacketar committed:
Merge branch 'master' into feature/tests
Everything compiles except the gridftp plugin.  This does have
the name-mangling changes applied.  The plugin doesn't compile
even from a fresh clone.  There seem to be issues with finding
the globus packages and also incorrect number of args.

tacketar committed:
Properly set the *_FOUND variables for the Globus libs
I don't see how these FindGlobus*.cmake routines ever worked.
Calling FindPackage(XxxYyy REQUIRED) checks for a XxxYyy_FOUND
variable on return to determine success and NOT Xxx_Yyy_FOUND which
which is what was set in both those routines.  I made the minimal
change of Xxx_Yyy_FOUND->XxxYyy_FOUND for both routines.  The other
library and include paths still have the name separation, Xxx_Yyy_*.
This allows the packages to be be found and builds the library.

tacketar committed:
Fix a double free
